router/batchrouter/asyncdestinationmanager/yandexmetrica/yandexmetrica.go

```diff
@@ -0,0 +1,373 @@
+	ClientID any     `json:"ClientId"`
+	YclID    any     `json:"Yclid"`
+	UserID   any     `json:"UserId"`
+	Target   string  `json:"Target"`
+	DateTime string  `json:"DateTime"`
+	Price    float64 `json:"Price"`
+	Currency string  `json:"Currency"`
+}
+
+type yandexMetricaMessage struct {
+	Message yandexMetricaMessageBody `json:"message"`
+}
+
+type idStruct struct {
+	id         string
+	clientType string
+	headerName string
+}
+
+func (ym yandexMetricaMessageBody) ID() (idStruct, error) {
+	switch {
+	case ym.ClientID != nil:
+		return getID(ym.ClientID, "ClientId")
+	case ym.YclID != nil:
+		return getID(ym.YclID, "Yclid")
+	case ym.UserID != nil:
+		return getID(ym.UserID, "UserId")
+	default:
+		return idStruct{}, fmt.Errorf("no valid id found in message object")
+	}
+}
+
+func getID(id interface{}, headerName string) (idStruct, error) {
+	idString, ok := id.(string)
+	if !ok {
+		return idStruct{}, fmt.Errorf("non-string data for %s is not supported", headerName)
+	}
+	return idStruct{id: idString, clientType: idClientMap[headerName], headerName: headerName}, nil
+}
+
+type YandexMetricaBulkUploader struct {
+	logger          logger.Logger
+	Client          *http.Client
+	destinationInfo *oauthv2.DestinationInfo
+}
+
+func NewManager(destination *backendconfig.DestinationT, backendConfig backendconfig.BackendConfig) (*YandexMetricaBulkUploader, error) {
+	destinationInfo := &oauthv2.DestinationInfo{
+		Config:           destination.Config,
+		DefinitionConfig: destination.DestinationDefinition.Config,
+		WorkspaceID:      destination.WorkspaceID,
+		DefinitionName:   destination.DestinationDefinition.Name,
+		ID:               destination.ID,
+	}
+	yandexUploadManager := &YandexMetricaBulkUploader{
+		destinationInfo: destinationInfo,
+		logger:          logger.NewLogger().Child("batchRouter").Child("AsyncDestinationManager").Child("YandexMetrica").Child("YandexMetricaBulkUploader"),
+	}
+	cache := oauthv2.NewCache()
+	optionalArgs := &oauthv2httpclient.HttpClientOptionalArgs{
+		Logger:    yandexUploadManager.logger,
+		Augmenter: augmenter.YandexReqAugmenter,
+	}
+	originalHttpClient := &http.Client{Transport: &http.Transport{}}
+	// This client is used for uploading data to yandex metrica
+	yandexUploadManager.Client = oauthv2httpclient.NewOAuthHttpClient(
+		originalHttpClient,
+		oauthv2common.RudderFlowDelivery,
+		&cache,
+		backendConfig,
+		augmenter.GetAuthErrorCategoryForYandex,
+		optionalArgs,
+	)
+	return yandexUploadManager, nil
+}
+
+// return a success response for the poll request every time by default
+func (ym *YandexMetricaBulkUploader) Poll(pollInput common.AsyncPoll) common.PollStatusResponse {
+	return common.PollStatusResponse{}
+}
+
+// return a success response for the getUploadStats request every time by default
+func (ym *YandexMetricaBulkUploader) GetUploadStats(UploadStatsInput common.GetUploadStatsInput) common.GetUploadStatsResponse {
+	return common.GetUploadStatsResponse{}
+}
```

Comment:
Due to lack of context on my part:
Is this yet to be implemented?
Why are we returning a success response by default.

Reply:
We do not have support of polling at Yandex end, so to maintain the bulk upload flow, by default we return success.

Reply:
But what if it's a failure..?
This kind of sounds like an object storage destination then(if there's no polling involved)

Reply:
If it's a failure we are discarding the events. We can poll the data, but it takes almost 24 hours to process it, so we are going with this approach.